### ISSUE TITLE
Huge File transfer (>5GB) to AIX host through SSH key credentials is always getting hung and SocketTimeoutException is reported

### ISSUE THREAD

**Issue by satishtedla** [apache/mina-sshd#432]
### Version

2.5.0 & 2.5.1

### Bug description

Huge File transfer (>5GB) to AIX host through SSH key credentials is always getting hung and SocketTimeoutException is reported
- Hang with SSH Key credentials + AIX is unpredictable. At times, File transfer hangs after transferring 300MB and at times at 2GB, 4.9GB too.
- Application uses Apache Mina SSHD client (2.5.0)
- Packet size configured for file transfer - 2400 bytes
- During the hang, Below is the exception stack captured in logs:
	- ERROR - flush(ChannelOutputStream[ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/x.x.x.x:xx]] SSH_MSG_CHANNEL_DATA) failed (SocketTimeoutException) to wait for space of len=24576: waitForCondition(Window[client/remote](ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/.   xxx.x.x.x:xx])) timeout exceeded: 120000
    	2023-10-03 17:15:47,703 [1069:CA6C8BB6] ERROR - flush(ChannelOutputStream[ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/xxx.x.x.x:xx]] SSH_MSG_CHANNEL_DATA) wait for space len=24576 exception details
	java.net.SocketTimeoutException: waitForCondition(Window[client/remote](ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/xxx.x.x.x:xx])) timeout exceeded: 120000
	at org.apache.sshd.common.channel.Window.waitForCondition(Window.java:307)
	at org.apache.sshd.common.channel.Window.waitForSpace(Window.java:255)
	at org.apache.sshd.common.channel.ChannelOutputStream.flush(ChannelOutputStream.java:197)
	at org.apache.sshd.common.channel.ChannelOutputStream.write(ChannelOutputStream.java:134)
	............................................................................................................
	............................................................................................................
	............................................................................................................
	at java.lang.Thread.run(Thread.java:825)
+ Tried to increase the sshd channel WAIT_FOR_SPACE_TIMEOUT from default 30 secs upto 10 mins but it did not help. Post the configured timeout, Above exception was reported.
      PropertyResolverUtils.updateProperty(sshClient, ChannelOutputStream.WAIT_FOR_SPACE_TIMEOUT, 120000);

### Actual behavior

File transfer fails with a SocketTimeoutException

### Expected behavior

File transfer must succeed with out any errors

### Relevant log output

```Shell
- Application uses Apache Mina SSHD client (2.5.0)
- Packet size configured for file transfer - 2400 bytes
- During the hang, Below is the exception stack captured in logs:
	- ERROR - flush(ChannelOutputStream[ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/x.x.x.x:xx]] SSH_MSG_CHANNEL_DATA) failed (SocketTimeoutException) to wait for space of len=24576: waitForCondition(Window[client/remote](ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/.   xxx.x.x.x:xx])) timeout exceeded: 120000
    	2023-10-03 17:15:47,703 [1069:CA6C8BB6] ERROR - flush(ChannelOutputStream[ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/xxx.x.x.x:xx]] SSH_MSG_CHANNEL_DATA) wait for space len=24576 exception details
	java.net.SocketTimeoutException: waitForCondition(Window[client/remote](ChannelExec[id=195, recipient=0]-ClientSessionImpl[xxxx@localhost/xxx.x.x.x:xx])) timeout exceeded: 120000
	at org.apache.sshd.common.channel.Window.waitForCondition(Window.java:307)
	at org.apache.sshd.common.channel.Window.waitForSpace(Window.java:255)
	at org.apache.sshd.common.channel.ChannelOutputStream.flush(ChannelOutputStream.java:197)
	at org.apache.sshd.common.channel.ChannelOutputStream.write(ChannelOutputStream.java:134)
	.............
        .............
        .............
	at java.lang.Thread.run(Thread.java:825)
+ Tried to increase the sshd channel WAIT_FOR_SPACE_TIMEOUT from default 30 secs upto 10 mins but it did not help. Post the configured timeout, Above exception was reported.
      PropertyResolverUtils.updateProperty(sshClient, ChannelOutputStream.WAIT_FOR_SPACE_TIMEOUT, 120000);
```

### Other information

Need your help/inputs on the possible issue.Is it a known issue? Can there be a workaround to prevent the hang? Please comment. Thanks!

**Comment by satishtedla** [comment]
Dear Team: We would really appreciate if this issue could be progressed. One of the Oracle Enterprise Manager(OEM) Customers is hitting this error consistently. We will be happy to explain the issue, share a few relevant data points. Thanks!

**Comment by tomaswolf** [comment]
We do not have the resources to investigate problems with old versions. Try with Apache MINA sshd 2.11.0.

There have been problems in past versions in key re-exchanges that could lead to hangs. AFAIK, all such problems are resolved in 2.11.0.
